tests/run-make/rustdoc-target-spec-json-path/rmake.rs

```diff
@@ -0,0 +1,15 @@
+// Test that rustdoc will properly canonicalize the target spec json path just like rustc.
+
+use run_make_support::{rustc, rustdoc, tmp_dir};
+
+fn main() {
+    let out_dir = tmp_dir().join("rustdoc-target-spec-json-path");
+    rustc().crate_type("lib").input("dummy_core.rs").target("target.json").run();
+    rustdoc()
+        .input("my_crate.rs")
+        .output(out_dir)
+        .arg("-L")
+        .arg(tmp_dir())
```

Comment:
I like

```rs
/// Add a directory to the library search path.
pub fn library_search_path<P: AsRef<Path>>(&mut self, path: P) -> &mut Self {
    self.cmd.arg("-L");
    self.cmd.arg(path.as_ref());
    self
}
```

introduced in https://github.com/rust-lang/rust/pull/125031 for `rustc`. Maybe we should add it for `rustdoc` too?

See https://github.com/rust-lang/rust/pull/125031/files#r1598288585 for discussion.

Reply:
Oh, it's convenient! Adding it to rustdoc then. :)